build.gradle

```diff
@@ -279,6 +295,78 @@ subprojects {
       exceptionFormat = testExceptionFormat
     }
     logTestStdout.rehydrate(delegate, owner, this)()
+
+    afterTest { TestDescriptor desc , TestResult result ->
+      if (TestResult.ResultType.FAILURE == result.resultType) {
+        logger.warn("Found failing test $desc.parent - $desc.name, trying again in next turn of test execution")
+        failedTestsInFirstTestRun << desc
+      }
+    }
+    testResultsDir = reports.junitXml.destination
+
+    ignoreFailures = shouldFilterFlakyTests
+  }
+
+  task testWithFlakyRetry(type: Test, dependsOn: test) {
+    maxParallelForks = userMaxForks ?: Runtime.runtime.availableProcessors()
+
+    minHeapSize = "256m"
+    maxHeapSize = "2048m"
+
+    testLogging {
+      events = userTestLoggingEvents ?: testLoggingEvents
+      showStandardStreams = userShowStandardStreams ?: testShowStandardStreams
+      exceptionFormat = testExceptionFormat
+    }
+
+    def oldResultsToOverwite = []
+
+    doFirst {
+      if (!failedTestsInFirstTestRun.isEmpty()) {
+        logger.warn("Executing failed tests again to filter out flaky tests")
```

Comment:
I see the rest of this file uses `println` to print statements. Are we moving away from that and do you know if there is any reason to prefer one over the other?

Reply:
Well there are 5 `println` statements in the file so I didn't know this was the convention :).
Anyway I'd rather use logging just because of the general advantages that logging has over `println`.

Reply:
Fair point :D. Makes sense to me